tests/publiccloud/registercloudguest.pm

```diff
@@ -0,0 +1,61 @@
+# SUSE's openQA tests
+#
+# Copyright 2022 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Package: cloud-regionsrv-client
+# Summary: Test system (re)registration
+#
+# Maintainer: <qa-c@suse.de>
+
+use Mojo::Base 'publiccloud::basetest';
+use version_utils;
+use registration;
+use warnings;
+use testapi;
+use strict;
+use utils;
+use publiccloud::utils;
+
+sub run {
+    my ($self, $args) = @_;
+
+    # Preserve args for post_fail_hook
+    $self->{instance} = $args->{my_instance};
+
+    # Create $instance to make the code easier to read
+    my $instance = $args->{my_instance};
+    my $regcode = (is_byos()) ? '-r ' . get_required_var('SCC_REGCODE') : '';
+
+    select_host_console();    # select console on the host, not the PC instance
+
+    $instance->run_ssh_command(cmd => '! sudo SUSEConnect -d', quiet => 0) if (is_byos());
+    $instance->run_ssh_command(cmd => 'sudo registercloudguest --clean', quiet => 0);
+    $instance->run_ssh_command(cmd => '! sudo zypper lr', timeout => 600);
+    $instance->run_ssh_command(cmd => '! sudo ls /etc/zypp/credentials.d/*', quiet => 0);
+    if (is_byos()) {
+        $instance->run_ssh_command(cmd => 'sudo SUSEConnect --version');
+        $instance->run_ssh_command(cmd => "sudo SUSEConnect $regcode");
+        my $remote = $args->{my_instance}->username . '@' . $args->{my_instance}->public_ip;
+        #register_addon($remote, 'pcm');
```

Comment:
Leftover?

Reply:
This is commented because of the [bsc#1198080](https://bugzilla.suse.com/show_bug.cgi?id=1198080)

Reply:
soft-failure than ?

Reply:
The test dies in `registration::ssh_remove_suseconnect_product`. It is too complicated to softfailure there. I will remove it for now and mention it in the progress ticket.